allennlp/common/testing/distributed_test_context_manager.py

```diff
@@ -0,0 +1,88 @@
+import datetime
+from typing import List
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+
+from allennlp.common.checks import check_for_gpu
+
+
+class DistributedTestContextManager:
+    """
+    This context manager is for simulating a distributed environment.
+
+    # Parameters
+
+    device_ids: `List[int]`
+        List of devices. There need to be at least 2 devices. Default is [-1, -1].
+
+    # Usage
+
+    ```
+    python
+    with DistributedTestContextManager(devices_id) as test_this_function:
+        test_this_function(your_distributed_func, *args)
```

Comment:
I strongly disagree that this is cleaner as a context manager, even if there was cleanup. You could restructure this so you need only the second line from your example:
```Python
test_distributed_function(devices_id, your_distributed_func, *args)
```
That's all it takes! The context manager adds nothing.